yoyodyne/models/lstm.py

```diff
@@ -166,38 +166,24 @@ def beam_decode(
         encoder_out: torch.Tensor,
         encoder_mask: torch.Tensor,
         beam_width: int,
-        n: int = 1,
-        return_confidences: bool = False,
-    ) -> Union[Tuple[torch.Tensor, torch.Tensor], torch.Tensor]:
-        """Beam search with beam_width.
+        n: int = defaults.N,
+    ) -> Tuple[torch.Tensor, torch.Tensor]:
+        """Beam search with beam width.
 
         Note that we assume batch size is 1.
-
-        Args:
```

Comment:
It is unclear to me why we would remove the method signature here. Is this typical when inheriting a method from an interface?

Reply:
I had the same question so I decided to follow the [PEP 257](https://peps.python.org/pep-0257/). 

> If a class subclasses another class and its behavior is mostly inherited from that class, its docstring should mention this and summarize the differences. Use the verb “override” to indicate that a subclass method replaces a superclass method and does not call the superclass method; use the verb “extend” to indicate that a subclass method calls the superclass method (in addition to its own behavior).

Anyway, I added some more information to the docstring.